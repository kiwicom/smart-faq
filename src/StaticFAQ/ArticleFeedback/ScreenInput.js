// @flow

import * as React from 'react';
import css from 'styled-jsx/css';
import { Heading, Text, Button } from '@kiwicom/orbit-components';
import { Close } from '@kiwicom/orbit-components/lib/icons';

import { Box } from '../../common';
import createComment from '../../mutations/CreateCommentMutation';
import screensList from './screensList';

type Props = {|
  changeScreen: (nextScreen: string) => void,
  articleId: string,
|};
type State = {|
  comment: string,
|};

const style = css`
  div.question {
    margin-bottom: 4px;
    margin-top: 8px;
  }
  div.inputArea textarea {
    width: 100%;
    height: 72px;
    border-radius: 3px;
    background-color: #ffffff;
    border: solid 1px #bac7d5;
    resize: none;
  }
  div.button {
    display: flex;
    justify-content: flex-end;
    margin-top: 19px;
  }
  div.close-icon {
    position: absolute;
    top: 8px;
    right: 8px;
    cursor: pointer;
  }
  div.inputArea textarea {
    padding: 12px 16px;
    font-size: 14px;
  }
  @media only screen and (max-width: 812px) {
    div.inputArea textarea {
      width: 100%;
    }
  }
`;

class ScreenInput extends React.Component<Props, State> {
  state = {
    comment: '',
  };
  handleChange = (e: SyntheticInputEvent<HTMLInputElement>) => {
    this.setState({ comment: e.target.value });
  };
  handleSubmit = (e: SyntheticEvent<HTMLFormElement>) => {
    const { changeScreen, articleId } = this.props;
    e.preventDefault();
    createComment(
      articleId,
      this.state.comment,
      () => changeScreen(screensList.THANK_YOU),
      () => changeScreen(screensList.ERROR),
    );
  };
  closeScreen = () => {
    this.props.changeScreen(screensList.INITIAL);
  };
  render() {
    return (
      <Box
        border="none"
        padding="40px 24px 24px 24px"
        borderRadius="4px"
        backgroundColor="#f5f7f9"
      >
        <form onSubmit={this.handleSubmit}>
          <div
            className="close-icon"
            onClick={this.closeScreen}
            onKeyUp={null}
            tabIndex="-1"
            role="button"
          >
            <Close customColor="#bac7d5" size="small" />
          </div>
          <Heading weight="bold" size="small">
            Your feedback helps us improve.
          </Heading>
          <div className="question">
            <Text>What problem were you trying to solve?</Text>
          </div>
          <div className="inputArea">
            <textarea data-gramm_editor="false" />
          </div>
          <div className="button">
            <Button title="Submit" onClick={() => {}} width={100} />
          </div>
        </form>
        <style jsx>{style}</style>
      </Box>
    );
  }
}

export default ScreenInput;
