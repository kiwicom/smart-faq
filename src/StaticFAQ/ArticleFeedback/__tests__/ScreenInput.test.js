// @flow

import * as React from 'react';
import { shallow } from 'enzyme';

import ScreenInput from '../ScreenInput';

const ch = (sc: string) => {}; //eslint-disable-line

describe('SignIn', () => {
  const component = shallow(
    <ScreenInput changeScreen={ch} articleId="WE4rT" />,
  );
  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
