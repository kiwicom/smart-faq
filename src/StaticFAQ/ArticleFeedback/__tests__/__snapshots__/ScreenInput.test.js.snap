// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenInput should match snapshot 1`] = `
<Box
  backgroundColor="#f5f7f9"
  border="none"
  borderRadius="4px"
  padding="40px 24px 24px 24px"
>
  <form
    className="jsx-982984599"
    onSubmit={[Function]}
  >
    <div
      className="jsx-982984599 close-icon"
      onClick={[Function]}
      onKeyUp={null}
      role="button"
      tabIndex="-1"
    >
      <Close
        customColor="#bac7d5"
        size="small"
      />
    </div>
    <Heading
      element="h1"
      size="small"
      weight="bold"
    >
      Your feedback helps us improve.
    </Heading>
    <div
      className="jsx-982984599 question"
    >
      <Text
        align="left"
        element="p"
        italic={false}
        size="normal"
        type="primary"
        uppercase={false}
        weight="regular"
      >
        What problem were you trying to solve?
      </Text>
    </div>
    <div
      className="jsx-982984599 inputArea"
    >
      <textarea
        className="jsx-982984599"
        data-gramm_editor="false"
      />
    </div>
    <div
      className="jsx-982984599 button"
    >
      <Button
        block={false}
        disabled={false}
        loading={false}
        onClick={[Function]}
        onlyIcon={false}
        size="normal"
        title="Submit"
        type="primary"
        variation="filled"
        width={100}
      />
    </div>
  </form>
  <JSXStyle
    css="div.question.jsx-982984599{margin-bottom:4px;margin-top:8px;}div.inputArea.jsx-982984599 textarea.jsx-982984599{width:100%;height:72px;border-radius:3px;background-color:#ffffff;border:solid 1px #bac7d5;}div.button.jsx-982984599{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:end;-webkit-justify-content:flex-end;-ms-flex-pack:end;justify-content:flex-end;margin-top:19px;}div.close-icon.jsx-982984599{position:absolute;top:8px;right:8px;cursor:pointer;}div.inputArea.jsx-982984599 textarea.jsx-982984599{padding:12px 16px;font-size:14px;}@media only screen and (min-width:320px) and (max-width:480px){div.inputArea.jsx-982984599 textarea.jsx-982984599{width:100%;}}"
    styleId="982984599"
  />
</Box>
`;
